File: 896.monotonic-array/896.monotonic-array.java

/*
 * @lc app=leetcode id=896 lang=java
 *
 * [896] Monotonic Array
 */

// @lc code=start
class Solution {
    public boolean isMonotonic(int[] nums) {
        if (nums.length < 3) {
            return true;
        }
        
        int minValue = nums[0];
        int maxValue = nums[0];
        boolean minCheck = true;
        boolean maxCheck = true;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= minValue) {
                minValue = nums[i];
            } else {
                minCheck = false;
            }
            
            if (nums[i] >= maxValue) {
                maxValue = nums[i];
            } else {
                maxCheck = false;
            }
            
            if (!minCheck && !maxCheck) {
                return false;
            }
        }
        
        return true;
    }
}

// @lc code=end

File: 896.monotonic-array/896.monotonic-array.py
#
# @lc app=leetcode id=896 lang=python3
#
# [896] Monotonic Array
#

# @lc code=start
from typing import List


class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return True
        
        minValue = nums[0]
        maxValue = nums[0]
        minCheck = True
        maxCheck = True
        
        for i in range(len(nums)):
            if nums[i] <= minValue:
                minValue = nums[i]
            else:
                minCheck = False
            
            if nums[i] >= maxValue:
                maxValue = nums[i]
            else:
                maxCheck = False
            
            if not minCheck and not maxCheck:
                return False
        
        return True
# @lc code=end

